File: src/models/image.model.js
import { Schema, model } from "mongoose";
import mongooseAggregatePaginate from "mongoose-aggregate-paginate-v2";

const imageSchema = new Schema({
  publicId: {
    type: String,
    required: true,
  },
  secureUrl: {
    type: String,
    required: true,
  },
  height: {
    type: Number,
    required: true,
  },
  width: {
    type: Number,
    required: true,
  },
  transformationUrl: {
    type: [
      {
        secureUrl: {
          type: String,
          required: true,
        },
      },
    ],
  },
  author: {
    type: Schema.Types.ObjectId,
    ref: "User",
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

imageSchema.plugin(mongooseAggregatePaginate);

export const Image = model("Image", imageSchema);

File: frontend/flutter/lib/model/image_model.dart
class ImageModel {
  String id;
  String publicId;
  String secureUrl;
  int height;
  int width;
  String author;
  DateTime createdAt;

  ImageModel(
      {required this.id,
      required this.publicId,
      required this.secureUrl,
      required this.height,
      required this.width,
      required this.author,
      required this.createdAt});

  Map<String, dynamic> toMap() {
    return {
      "id": id,
      "publicId": publicId,
      "secureUrl": secureUrl,
      "height": height,
      "width": width,
      "author": author,
      "createdAt": createdAt,
    };
  }

  factory ImageModel.fromMap(Map<String, dynamic> map) {
    return ImageModel(
        id: map["_id"],
        publicId: map['publicId'],
        secureUrl: map['secureUrl'],
        height: map['height'],
        width: map['width'],
        author: map['author'],
        createdAt: DateTime.parse(map['createdAt']));
  }
}
